core/src/main/java/hudson/model/User.java

```diff
@@ -713,8 +714,9 @@ public static boolean isIdOrFullnameAllowed(@CheckForNull String id) {
         if (id == null || StringUtils.isBlank(id)) {
             return false;
         }
+        final String trimmedId = id.trim();
         for (String invalidId : ILLEGAL_PERSISTED_USERNAMES) {
-            if (id.equalsIgnoreCase(invalidId))
+            if (trimmedId.equalsIgnoreCase(invalidId))
```

Comment:
[JavaDoc](https://docs.oracle.com/javase/7/docs/api/java/lang/String.html#equalsIgnoreCase(java.lang.String))

> Two characters c1 and c2 are considered the same ignoring case if at least one of the following is true:
* The two characters are the same (as compared by the == operator)
* Applying the method [`Character.toUpperCase(char)`](https://docs.oracle.com/javase/7/docs/api/java/lang/Character.html#toUpperCase(char)) to each character produces the same result
* Applying the method [`Character.toLowerCase(char)`](https://docs.oracle.com/javase/7/docs/api/java/lang/Character.html#toLowerCase(char)) to each character produces the same result

So yes - this is "safe" for different locales - but not correct.

For example a lower case 'i' will always be compared to 'I', however in some locales the lower case i maps to a different character.   So I think the characters we use for system accounts are 1:1 mapped for the locales I am aware of - but it seems a bit prone to error.

Even specifying the root a locale and lower casing is not what is really wanted here either - we need to do the match based on what the user would see as well - and that means locale specific and not known when the User is created.